test/tools/libtesteth/TestHelper.cpp

```diff
@@ -242,21 +243,51 @@ void replaceLLLinState(json_spirit::mObject& _o)
 	}
 }
 
-std::vector<boost::filesystem::path> getJsonFiles(boost::filesystem::path const& _dirPath, std::string const& _particularFile)
+template<class T>
+std::set<T> removeRepeatElementsFromSet(std::set<T> const& _set)
+{
+	std::set<T> uniqueSet;
+	for(auto const& element: _set)
+	{
+		bool found = false;
+		for(auto const& uniqueElement: uniqueSet)
+		{
+			if (element == uniqueElement)
+			{
+				found = true;
+				break;
+			}
+		}
+		if(!found)
+			uniqueSet.insert(element);
+	}
+	return uniqueSet;
+}
+
+std::vector<boost::filesystem::path> getFiles(boost::filesystem::path const& _dirPath, std::set<string> const _extentionMask, std::string const& _particularFile)
 {
 	vector<boost::filesystem::path> jsonFiles;
+	std::set<string> extentions = removeRepeatElementsFromSet(_extentionMask);
 	if (!_particularFile.empty())
 	{
-		boost::filesystem::path file = _dirPath / (_particularFile + ".json");
-		if (boost::filesystem::exists(file))
-			jsonFiles.push_back(file);
+		for(auto const& ext: extentions)
+		{
+			boost::filesystem::path file = _dirPath / (_particularFile + ext);
+			if (boost::filesystem::exists(file))
+				jsonFiles.push_back(file);
+		}
 	}
 	else
 	{
 		using Bdit = boost::filesystem::directory_iterator;
 		for (Bdit it(_dirPath); it != Bdit(); ++it)
-			if (boost::filesystem::is_regular_file(it->path()) && it->path().extension() == ".json")
-					jsonFiles.push_back(it->path());
+		{
+			for(auto const& ext: extentions)
```

Comment:
This code looks similar to the loop above. Can you refactor it to have single loop of extensions?

Reply:
I think this two for loop's could be moved upwards and combined.